plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +49,15 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
+	for _, predicates := range f.FailedPredicates {
```

Comment:
Yeah I get that, just denoting that internally we are storing f.FailedPredicates which will still be (N).